codegen/sdk-codegen/build.gradle.kts

```diff
@@ -35,8 +38,53 @@ tasks.create<SmithyBuild>("buildSdk") {
     addRuntimeClasspath = true
 }
 
+// Generates a smithy-build.json file by creating a new projection for every
+// JSON file found in aws-models/. The generated smithy-build.json file is
+// not committed to git since it's rebuilt each time codegen is performed.
+tasks.register("generate-smithy-build") {
+    doLast {
+        val projectionsBuilder = Node.objectNodeBuilder()
+        val modelsDirProp: String by project
+        val models = project.file(modelsDirProp);
+
+        fileTree(models).filter { it.isFile }.files.forEach { file ->
+            val model = Model.assembler()
+                    .addImport(file.absolutePath)
+                    // Grab the result directly rather than worrying about checking for errors via unwrap.
+                    // All we care about here is the service shape, any unchecked errors will be exposed
+                    // as part of the actual build task done by the smithy gradle plugin.
+                    .assemble().result.get();
+            val services = model.shapes(ServiceShape::class.javaObjectType).sorted().toList();
+            if (services.size != 1) {
+                throw Exception("There must be exactly one service in each aws model file, but found " +
+                        "${services.size} in ${file.name}: ${services.map { it.id }}");
+            }
+            val service = services[0]
+            var (sdkId, version, remaining) = file.name.split(".")
+            sdkId = sdkId.replace("-", "").toLowerCase();
```

Comment:
Should spaces be stripped out as well?  ServiceIds aren't supposed to contain white space, but wasn't sure if Smithy validated this. Several of the old c2j modules contain spaces within their serviceId values.

Reply:
This is based on the filename, which already has any set of consecutive whitespace chars replaced with `-`